server/src/main/java/org/elasticsearch/cluster/SnapshotsInProgress.java

```diff
@@ -115,8 +118,8 @@ public static Entry startedEntry(Snapshot snapshot, boolean includeGlobalState,
     public static Entry startClone(Snapshot snapshot, SnapshotId source, List<IndexId> indices, long startTime,
                                    long repositoryStateId, Version version) {
         return new SnapshotsInProgress.Entry(snapshot, true, false, State.STARTED, indices, Collections.emptyList(),
-                startTime, repositoryStateId, ImmutableOpenMap.of(), null, Collections.emptyMap(), version, source,
-                ImmutableOpenMap.of());
+            Collections.emptyList(), startTime, repositoryStateId, ImmutableOpenMap.of(), null, Collections.emptyMap(), version, source,
```

Comment:
This seems like a good point. Snapshot cloning is relatively new and I don't know much about it - if that's the case though, why are data streams `Collections.emptyList()` here? Shouldn't they be copied as well?

Reply:
:facepalm: yet another datastream curve-ball :) This should be fixed as well. Let's do what you did here then maybe and add a TODO for this. I don't think it's critical to fix in this PR since it's broken already for datastreams. Thanks for noticing!